src/WebJobs.Script/ServiceManagers/IManagedHostedService.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.ServiceManagers
+{
+    public interface IManagedHostedService
+    {
+        Task StopServicesAsync(CancellationToken cancellationToken);
```

Comment:
From RpcInitializationService's perspective, it stops a single service. I was thinking , what if there are other IHostedService(s) that implement this which could potentially stop multiple services? Also, if it's StopAsync, were you thinking of StopAsync() without a cancellationToken? Wouldn't it be better to have a different method name but with access to CancellationToken rather than have StopAsync() without one..?

Reply:
No, I was thinking still taking a cancellation token. This will require an explicit implementation (if you don’t go with the other suggestion).

For the implementer of the interface, this is always a call to stop that specific service, so the singular version would be appropriate. This is moot if we rename with the suggested name, though